paddle/fluid/framework/ir/multi_devices_graph_pass/all_reduce_deps_pass.cc

```diff
@@ -131,7 +131,7 @@ class AllReduceDepsPass : public ir::Pass {
            auto right_in_vars =
                details::DynamicCast<details::VarHandle>(right->Inputs());
            PADDLE_ENFORCE_GT(left_in_vars.size(), 0);
-           PADDLE_ENFORCE_EQ(left_in_vars.size(), right_in_vars.size());
```

Comment:
The size of `left_in_vars ` and `right_in_vars ` may be different when the `fuse_all_reduce_ops` is enabled.

Reply:
I was surprised that I didn’t encounter this bug before.